dbus-java-core/src/main/java/org/freedesktop/dbus/StructHelper.java

```diff
@@ -137,7 +138,7 @@ public static <T extends Struct> T createStructFromVariant(Variant<?> _variant,
         }
 
         if (_variant.getType() instanceof DBusStructType && _variant.getValue() instanceof Object[]) {
-            Class<?>[] argTypes = Arrays.stream((Object[]) _variant.getValue()).map(Object::getClass).toArray(size -> new Class<?>[size]);
+            Class<?>[] argTypes = Arrays.stream((Object[]) _variant.getValue()).map(Object::getClass).toArray((IntFunction<Class<?>[]>) Class[]::new);
```

Comment:
This can be simplified to `Class<?>[] argTypes = Arrays.stream((Object[]) _variant.getValue()).map(Object::getClass).toArray(Class<?>[]::new);`